src/plugins/expressions/common/expression_functions/specs/kibana_context.ts

```diff
@@ -74,13 +73,21 @@ export const kibanaContext = (): ExpressionFunctionKibanaContext => ({
       }),
     },
   },
-  async fn(input, args) {
+
+  async fn(input, args, { getSavedObject }) {
```

Comment:
instead of polluting the handlers lets pass this down to the function on the time of setup, either:
- use the pattern we already use in many places: `setGetSavedObject ()`(in the plugin setup method) and `getGetSavedObject()` inside the function.
- or wrap the function definition in a function and pass getSavedObject to it.
i prefer the first one as it allows you to pass in start contracts (even if you register in setup method) of core and our services.

Reply:
ignore this comment, we talked on zoom and saved object getter could be different with every execution on the server, as we need to scope it to the user. This means we need to save it on execution context. the same will be the case with uiSettingsClient, search service and other services.